lightkurve/lightcurve.py

```diff
@@ -103,44 +103,81 @@ def _validate_array(self, arr, name='array'):
         return arr
 
     def __getitem__(self, key):
-        copy_self = copy.copy(self)
+        copy_self = self.copy()
         copy_self.time = self.time[key]
         copy_self.flux = self.flux[key]
         copy_self.flux_err = self.flux_err[key]
         return copy_self
 
+    def __len__(self):
+        return len(self.time)
+
     def __add__(self, other):
-        copy_self = copy.copy(self)
-        copy_self.flux = copy_self.flux + other
-        return copy_self
+        newlc = self.copy()
+        if isinstance(other, LightCurve):
+            if len(self) != len(other):
+                raise ValueError("Cannot add LightCurve objects because "
+                                 "they do not have equal length ({} vs {})."
+                                 "".format(len(self), len(other)))
+            if np.any(self.time != other.time):
+                warnings.warn("Two LightCurve objects with inconsistent time "
+                              "values are being added.")
+            newlc.flux = self.flux + other.flux
```

Comment:
Can you add some more catches here? e.g. what if I try to add a `TessLightCurve` and a `KeplerLightCurve`? What if I want to add two light curves with the `metadata` dictionary filled with values? What happens to the attributes of the light curve? (e.g. column, row, centroid)

Reply:
I'd say make it `np.allclose()` for the time values, since times can be slightly spatially dependent due to the differences in BKJD conversion, and instead of a warning, raise an error.  I think its fair to only support adding of time-aligned lightcurves.  That change will address the Kepler/Tess lightcurves.

The issue with how to handle metadata is tricky!  We could elect to pass on the metadata from only one of the lightcurves.  But which one?  How to deal with the reverse?